### PR TITLE
WT-2667 Combine compile and unit-test tasks in EVG on all platforms

### PR DESCRIPTION
**Pull request by daveh86**

### CODE REVIEW

**Comment by daveh86** [comment]
@markbenvenuto, I'm combining the Posix and Windows compile and unit-test tasks here so we have only the one compile task show for all tests and we can save the binaries, etc. Can you take a look? This is currently passing on EVG

**Comment by daveh86** [comment]
retest this please

**Comment by daveh86** [comment]
recheck this please

**Comment by daveh86** [comment]
retest this please

**Comment by daveh86** [comment]
retest this please

File: test/mciproject.yml [hunk 1]
```diff
@@ -39,15 +39,18 @@ tasks:
...
           script: |
             set -o errexit
             set -o verbose
-
-            ./reconf
-            ${configure_env_vars|} ../configure --enable-diagnostic --enable-python --enable-zlib --enable-strict
-            ${make_command|make} ${smp_command|}
-            ${make_command|make} check
+            if [ "Windows_NT" = "$OS" ]; then
+              scons.bat --enable-python=c:\\swigwin-3.0.2\\swig.exe ${smp_command|}
```
Comment (markbenvenuto) [inline test/mciproject.yml]: specify `--enable-diagnostic` to match Posix

**Comment by markbenvenuto** [comment]
LGTM with one minor suggestion.

**Comment by daveh86** [comment]
@agorrod, with Mark's LGTM, are you able to have a look at this?

**Comment by agorrod** [comment]
Thanks @daveh86 lgtm - I'll merge.

**Pull request merged**